rendering.js

```diff
@@ -0,0 +1,39 @@
+// __This is the main render function which delegates control to the more-specific render functions.__
+function renderGame(state) {
+    // This would be the mode the game starts in
```

Comment:
I always use 2 space tabs, but I see you and @timotiousPrime use 4space, is it okay to have both?

Reply:
It's ok to have both, but it will probably get annoying. Have you used `prettier`? I've been using it in all my projects the last 2-3 years and it's been really nice. It's an opinionated code formatter, which essentially means you don't have to think about formatting ever again if you set it up on-save/commit. 

**Suggestion:** If you guys have no problem with auto-formatting, I could set up a vscode project-level settings in the repo so all our code becomes consistently styled? @timotiousPrime

Reply:
I've been using prettier, I haven't made any changes to it since installing it. I got no problem with auto-formatting so if you want to do that I'm up for it

Reply:
I do use prettier, I don't think it's gonna be a problem the the tabs thing

Reply:
Cool, I'll set up a shared prettier conf and we can all stick to the same format